test/gtest/ucp/test_ucp_wireup.cc

```diff
@@ -1570,3 +1570,94 @@ UCS_TEST_P(test_ucp_wireup_keepalive, attr) {
 }
 
 UCP_INSTANTIATE_TEST_CASE(test_ucp_wireup_keepalive)
+
+class test_ucp_address_v2 : public test_ucp_wireup {
+public:
+    static void get_test_variants(std::vector<ucp_test_variant>& variants)
+    {
+        add_variant_with_value(variants, UCP_FEATURE_TAG,
+                               TEST_TAG | WORKER_ADDR_V2, "tag");
+        add_variant_with_value(variants, UCP_FEATURE_TAG,
+                               TEST_TAG | WORKER_ADDR_V2 | UNIFIED_MODE,
+                               "tag,unified");
+    }
+
+    void check_fp_values(double unpacked, double original, double convert)
+    {
+        double max_error = original / pow(2, _UCS_FP8_MANTISSA_BITS);
+        EXPECT_NEAR(original, unpacked, max_error);
+    }
+};
+
+UCS_TEST_P(test_ucp_address_v2, pack_iface_attrs) {
+    ucs_status_t status;
+    size_t size;
+    void *buffer;
+
+    if (has_transport("tcp")) {
+        // On some systems TCP has very low BW and high latency, which would be
+        // unpacked by min/max values of the corresponding fp8 types
+        UCS_TEST_SKIP_R("skip checking tcp performance");
+    }
+
+    ucp_worker_h worker = sender().worker();
+
+    status = ucp_address_pack(worker, NULL, &ucp_tl_bitmap_max,
+                              UCP_ADDRESS_PACK_FLAGS_ALL, UCP_OBJECT_VERSION_V2,
+                              NULL, &size, &buffer);
+    ASSERT_UCS_OK(status);
+    ASSERT_TRUE(buffer != NULL);
+
+    ucp_unpacked_address unpacked_address;
+    status = ucp_address_unpack(worker, buffer, UCP_ADDRESS_PACK_FLAGS_ALL,
+                                &unpacked_address);
+    ASSERT_UCS_OK(status);
```

Comment:
if this assert fails, "buffer" will leak

Reply:
fixed, but this is an assert which aborts the test and it is used this way thru the whole file (and in other tests as well)